aten/src/ATen/native/mkldnn/xpu/detail/Matmul.cpp

```diff
@@ -263,4 +263,152 @@ sycl::event matmul(
+       at::native::onednn::get_onednn_dtype(mat2),
+       mat2.strides().vec()},
+      engine,
+      mat2.data_ptr());
+  dnnl::memory dst = at::native::onednn::make_onednn_memory(
+      {dst_dims,
+       at::native::onednn::get_onednn_dtype(result),
+       result.strides().vec()},
+      engine,
+      result.data_ptr());
+  bool with_bias = bias.has_value();
+  dnnl::memory::desc bias_desc = dnnl::memory::desc();
+  dnnl::memory onednn_bias;
+  if (with_bias) {
+    auto bias_value = bias.value();
+    if (bias_value.dim() == 1) {
+      TORCH_CHECK(
+          bias_value.size(0) == N || bias_value.size(0) == 1,
+          "matmul supports [n] or [1] when bias dim is 1 ...");
+      auto b_reshape = bias_value.reshape({1, bias_value.size(0)});
+      onednn_bias = at::native::onednn::make_onednn_memory(
+          {b_reshape.sizes().vec(),
+           at::native::onednn::get_onednn_dtype(b_reshape),
+           b_reshape.strides().vec()},
+          engine,
+          b_reshape.data_ptr());
+    } else {
+      onednn_bias = at::native::onednn::make_onednn_memory(
+          {bias_value.sizes().vec(),
+           at::native::onednn::get_onednn_dtype(bias_value),
+           bias_value.strides().vec()},
+          engine,
+          bias_value.data_ptr());
+    }
+  }
+
+  dnnl::primitive_attr op_attr = dnnl::primitive_attr();
+  if (scale_a.numel() == 1) {
+    op_attr.set_scales_mask(DNNL_ARG_SRC, 0);
+  } else {
+    // onednn 3.7 not support per token src scale, so use post mul work around
+    dnnl::post_ops po;
+    po.append_binary(
+        dnnl::algorithm::binary_mul,
+        {{scale_a.numel(), 1},
+         dnnl::memory::data_type::f32,
+         dnnl::memory::format_tag::ab});
+    op_attr.set_post_ops(po);
+  }
+  op_attr.set_scales_mask(DNNL_ARG_WEIGHTS, scale_b.numel() == 1 ? 0 : 1 << 1);
+  op_attr.set_scratchpad_mode(dnnl::scratchpad_mode::user);
+  // TODO: Remove this try/catch when oneDNN provides API to notify
+  // framework whether current platform can run FP8 primitives.
+  dnnl::matmul::primitive_desc primitive_desc;
+  try {
+    primitive_desc = with_bias ? dnnl::matmul::primitive_desc(
+                                     engine,
+                                     src.get_desc(),
+                                     weight.get_desc(),
+                                     onednn_bias.get_desc(),
+                                     dst.get_desc(),
+                                     op_attr)
+                               : dnnl::matmul::primitive_desc(
+                                     engine,
+                                     src.get_desc(),
+                                     weight.get_desc(),
+                                     dst.get_desc(),
+                                     op_attr);
+  } catch (dnnl::error& e) {
+    if (e.status == dnnl_unimplemented)
+      throw std::runtime_error("Onednn cannot create primitive.");
+    // on any other error just re-throw
+    throw;
```

Comment:
Which component should be responsible for catching the exception? Meanwhile, pls. revisie the error message to make it more informative.

Reply:
If the error is `not implementated`, XPU should raise NotImplemented as well.

Reply:
Meanwhile, it would be helpful to paste oneDNN issue links here if we have submitted issues to the oneDNN community.